// @aws-sdk/middleware-sdk-sqs/dist/cjs/send-message.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getSendMessagePlugin = exports.sendMessageMiddlewareOptions = exports.sendMessageMiddleware = void 0;
const util_hex_encoding_1 = require("@aws-sdk/util-hex-encoding");
const sendMessageMiddleware = (options) => (next) => async (args) => {
    const resp = await next({ ...args });
    const output = resp.output;
    const hash = new options.md5();
    hash.update(args.input.MessageBody || "");
    if (output.MD5OfMessageBody !== util_hex_encoding_1.toHex(await hash.digest())) {
        throw new Error("InvalidChecksumError");
    }
    return resp;
};
exports.sendMessageMiddleware = sendMessageMiddleware;
exports.sendMessageMiddlewareOptions = {
    step: "initialize",
    tags: ["VALIDATE_BODY_MD5"],
    name: "sendMessageMiddleware",
    override: true,
};
const getSendMessagePlugin = (config) => ({
    applyToStack: (clientStack) => {
        clientStack.add(exports.sendMessageMiddleware(config), exports.sendMessageMiddlewareOptions);
    },
});
exports.getSendMessagePlugin = getSendMessagePlugin;
